devel/lazarus/Portfile

```diff
@@ -0,0 +1,124 @@
+
+    reinplace "s|1\.\.15|1\.\.16|g" ${patchtarget}
+    reinplace "s|'/usr/share/fpcsrc'|'/usr/share/fpcsrc',\\
+    '${prefix}/share/fpcsrc'|g"      ${patchtarget}
+    reinplace "s|1\.\.8|1\.\.9|g" ${patchtarget}
+    reinplace "s|'/usr/share/lazarus',|'/usr/share/lazarus',\\
+    '${prefix}/share/lazarus',|g" ${patchtarget}
+
+# change the preferences directory (actually, not yet)
+#   reinplace "s|'\.lazarus'|'Library/Preferences/lazarus'|g" ${patchtarget}
+    reinplace "s|'/etc/lazarus'|'${prefix}/etc/lazarus'|g"    ${patchtarget}
+
+    reinplace "s|'/usr/local/bin/'+GetDefaultCompilerFilename);|'${prefix}/bin/'+GetDefaultCompilerFilename);\\
+  AddFilenameToList(List,'/usr/local/bin/'+GetDefaultCompilerFilename);|g" ${patchtarget}
+
+# adjust the default paths to MacPorts paths
+
+    set patchtarget ${worksrcpath}/tools/install/macosx/environmentoptions.xml
+
+    reinplace "s|/usr/local/bin/_PPCARCH_|${prefix}/bin/ppcx64|g" ${patchtarget}
+    reinplace "s|/usr/local/share|${prefix}/share|g"              ${patchtarget}
+    reinplace "s|/Developer/lazarus|${prefix}/share/lazarus|g"    ${patchtarget}
+
+    reinplace "s|<DebuggerFilename Value=\"/usr/bin/lldb\"/>|<DebuggerFilename Value=\"/usr/bin/lldb\">\\
+        <History Count=\"3\">\\
+          <Item1 Value=\"/usr/bin/lldb\"/>\\
+          <Item2 Value=\"/usr/bin/gdb\"/>\\
+          <Item3 Value=\"${prefix}/bin/gdb\"/>\\
+        </History>\\
+      </DebuggerFilename>|g" ${patchtarget}
+
+# fix path to X11
+    reinplace "s|-Fl/usr/X11R6/lib -Fl/sw/lib|\"-Fl${prefix}/lib -Fl${prefix}/lib/pango-ft219/lib -Fl/opt/X11/lib\"|g" ${worksrcpath}/ide/Makefile
+
+# some more fixes for nogui taking cocoa as a template
+# if cocoa or nogui get updated, the files in ${filespath} need to be updated manually
+
+    copy ${filespath}/nogui ${worksrcpath}/components/lclextensions/include/
+
+    copy ${worksrcpath}/components/virtualtreeview/units/cocoa         ${worksrcpath}/components/virtualtreeview/units/nogui
+    copy ${worksrcpath}/components/virtualtreeview/include/intf/cocoa/ ${worksrcpath}/components/virtualtreeview/include/intf/nogui
+
+}
+
+build.target       bigide
+build.post_args    OPT=-gl LCL_PLATFORM=cocoa
+use_parallel_build no
+
+destroot.post_args INSTALL_PREFIX=${destroot}${prefix}
+
+post-destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/etc/lazarus
+    xinstall -m 644 ${worksrcpath}/tools/install/macosx/environmentoptions.xml ${destroot}${prefix}/etc/lazarus
+
+# fix the symbolic links in the application bundle
+# and remove the questionable app bundle in Resources
+
+    set appContents ${destroot}${prefix}/share/lazarus/lazarus.app/Contents
+
+    delete                                  ${appContents}/Resources/startlazarus.app
+    delete                                  ${appContents}/MacOS/lazarus
+    ln ${prefix}/share/lazarus/lazarus      ${appContents}/MacOS/lazarus
+    ln ${prefix}/share/lazarus/startlazarus ${appContents}/MacOS/startlazarus
```

Comment:
These files don't actually exist in $prefix yet during the destroot phase, so you can't hardlink with those paths. If a symlink won't work, you need to add ${destroot} to the start of the paths because that's where the files are.

Reply:
The latest commit fixes this with symlinks.